Terminal.Gui/ConsoleDrivers/WindowsDriver.cs

```diff
@@ -773,34 +929,97 @@ public COLORREF (uint value)
 	[DllImport ("kernel32.dll", SetLastError = true)]
 	static extern Coord GetLargestConsoleWindowSize (
 	    IntPtr hConsoleOutput);
+
+	[DllImport ("kernel32.dll", SetLastError = true)]
+	public static extern bool SetConsoleTextAttribute (IntPtr hConsoleOutput, int wAttributes);
+
+	public bool SetTextAttribute (int wAttributes)
+	{
+		if (!SetConsoleTextAttribute (_screenBuffer, wAttributes)) {
+			var err = Marshal.GetLastWin32Error ();
+			if (err != 0) {
+				throw new System.ComponentModel.Win32Exception (err);
+			}
+		}
+
+		return true;
+	}
 }
 
 internal class WindowsDriver : ConsoleDriver {
-	WindowsConsole.ExtendedCharInfo [] _outputBuffer;
+	WindowsConsole.ExtendedRuneInfo [,] _outputBuffer;
 	WindowsConsole.SmallRect _damageRegion;
 
 	public WindowsConsole WinConsole { get; private set; }
 
 	public override bool SupportsTrueColor => RunningUnitTests || (Environment.OSVersion.Version.Build >= 14931 && _isWindowsTerminal);
 
 	readonly bool _isWindowsTerminal = false;
+	enum ConsoleHost {
+		AutomaticSelection,
+		WindowsConsoleHost,
+		WindowsTerminal,
+		NotConfigured
+	}
+	readonly ConsoleHost _consoleHost = ConsoleHost.NotConfigured;
 	WindowsMainLoop _mainLoopDriver = null;
 
 	public WindowsDriver ()
 	{
+		// TODO: if some other Windows-based terminal supports true color, update this logic to not
+		// force 16color mode (.e.g ConEmu which really doesn't work well at all).
+		_consoleHost = GetConsoleHost ();
+		_isWindowsTerminal = Environment.GetEnvironmentVariable ("WT_SESSION") != null ||
+			_consoleHost == ConsoleHost.AutomaticSelection || _consoleHost == ConsoleHost.WindowsTerminal;
+		if (!_isWindowsTerminal) {
+			Force16Colors = true;
+		}
+
 		if (Environment.OSVersion.Platform == PlatformID.Win32NT) {
-			WinConsole = new WindowsConsole ();
+			WinConsole = new WindowsConsole (_isWindowsTerminal);
 			// otherwise we're probably running in unit tests
 			Clipboard = new WindowsClipboard ();
 		} else {
 			Clipboard = new FakeDriver.FakeClipboard ();
 		}
+	}
 
-		// TODO: if some other Windows-based terminal supports true color, update this logic to not
-		// force 16color mode (.e.g ConEmu which really doesn't work well at all).
-		_isWindowsTerminal = Environment.GetEnvironmentVariable ("WT_SESSION") != null;
-		if (!_isWindowsTerminal) {
-			Force16Colors = true;
+	ConsoleHost GetConsoleHost ()
```

Comment:
This has nothing to do with Chinese rendering, right? Please remove it from this PR.

Reply:
Was the only way I could find to run on `Windows Terminal` or on `conhost` dependent of the Windows Terminal configuration.

Reply:
Was the most simple way instead of creating a lot of profile settings for now.

Reply:
Done

Reply:
@tig I really need this, otherwise I get the error bellow. This is needed for `conhost` restore the terminal defaults on exit (otherwise you will the last color used by the screen buffer). On Windows Terminal this code throw the exception because `csbi.srWindow.Bottom += 2;` cause a invalid value, since it use alternative screen. But I need this on `conhost` to adjust the size, otherwise the size will decrease always when you open and close a scenario. The only way I have to test both in this PR is using the `GetConsoleHost` method. If you later don't need this do whathever you want.

![image](https://github.com/gui-cs/Terminal.Gui/assets/13117724/7843ce6a-e2b0-46bf-83f4-f1fce949e04d)